### ISSUE TITLE
The previous and next buttons are shifted in time

### ISSUE THREAD

**Issue by AmazingBeu** [modoboa/modoboa-dmarc#35]
Exemple for this URL :  https://#########/dmarc/domains/1/?period=2019-4
The "previous" button redirect to 2019-4 and the "next" button to 2019-6.

This problem appear for 2019 and 2020, but not for 2018 or 2021.

**Comment by AmazingBeu** [comment]
I found this : 

In **templatetags/dmarc_tags.py** file, the week number sent will be incremented when you convert it to date format then to the ISO format .

exemple :
```
>>> import datetime
>>> period = "2019-11"
>>> current = datetime.datetime.strptime("{}-1".format(period), "%Y-%W-%w")
>>> print(current.isocalendar())
(2019, 12, 1)
```
but does not happen with 2018.

I found this workaround to make buttons work with an offset with the week number : 

```
@register.simple_tag
def next_period(period):
    """Return next period."""
    current = datetime.datetime.strptime("{}-1".format(period), "%Y-%W-%w")

    if int(period.split("-")[1]) is int(current.isocalendar()[1]):
        current += relativedelta.relativedelta(weeks=1)

    parts = current.isocalendar()
    return mark_safe("{}-{}".format(parts[0], parts[1]))


@register.simple_tag
def previous_period(period):
    """Return previous period."""
    current = datetime.datetime.strptime("{}-1".format(period), "%Y-%W-%w")

    if int(period.split("-")[1]) is not int(current.isocalendar()[1]):
        current += relativedelta.relativedelta(weeks=-1)

    current += relativedelta.relativedelta(weeks=-1)
    parts = current.isocalendar()
    return mark_safe("{}-{}".format(parts[0], parts[1]))
```

I hope this will help you solve the problem :)

**Comment by tonioo** [comment]
@AmazingBeu thank you. Could you please provide a PR ?

**Comment by PatTheMav** [comment]
Was just about to report this one, thank you for the fix. I see @tonioo just recently merged it, so hopefully 1.1.1 will be released soon.
